File: 45-shortest-palindrome/JavaScript/solution1.js
// Brute force solution:

// Time complexity: O(n^2)
// Space complexity: O(1)

function shortestPalindrome(str) {
    let mirrorLength = 0;
    for (let i = 0; i <= str.length; i++) {
        if (
            str.substring(0, i) ==
            str.substring(0, i).split('').reverse().join('')
        )
            mirrorLength = i;
    }
    return str.substring(mirrorLength).split('').reverse().join('') + str;
}

File: 22-jump-to-last-index/Python/solution1.py
# By using recursion:

# Time complexity: O(2^n)
# Space complexity: O(n)


def canJump(arr, i = 0):
  if i == len(arr)-1:
    return True
  for j in range(1, arr[i]+1):
    if canJump(arr, i+j):
      return True
  return False


